example/lib/main.dart

```diff
@@ -55,23 +102,25 @@ class _WidgetGalleryState extends State<WidgetGallery> {
 
   late final searchFieldController = TextEditingController();
 
-  final List<Widget> pages = [
-    CupertinoTabView(
-      builder: (_) => const ButtonsPage(),
-    ),
-    const IndicatorsPage(),
-    const FieldsPage(),
-    const ColorsPage(),
-    const Center(
-      child: MacosIcon(
-        CupertinoIcons.add,
-      ),
-    ),
-    const DialogsPage(),
-    const ToolbarPage(),
-    const SliverToolbarPage(),
-    const TabViewPage(),
-    const SelectorsPage(),
+  final List<Widget Function(bool)> pageBuilders = [
+    (bool isVisible) => CupertinoTabView(
+          builder: (_) => const ButtonsPage(),
+        ),
+    (bool isVisible) => const IndicatorsPage(),
+    (bool isVisible) => const FieldsPage(),
+    (bool isVisible) => const ColorsPage(),
+    (bool isVisible) => const Center(
+          child: MacosIcon(
+            CupertinoIcons.add,
+          ),
+        ),
+    (bool isVisible) => const DialogsPage(),
+    (bool isVisible) => const ToolbarPage(),
+    (bool isVisible) => SliverToolbarPage(
+          isVisible: isVisible,
+        ),
+    (bool isVisible) => const TabViewPage(),
+    (bool isVisible) => const SelectorsPage(),
```

Comment:
Can you please explain why it was necessary to make this change, as well as the change to the `IndexedStack`?

Reply:
Admittedly, this isn’t a very elegant solution, but I just couldn’t find a better one. You see, `SliverToolbarPage` contains a transparent sliver that uses a blurry backdrop filter. Since blurry backdrop filters and wallpaper tinting don’t mix, I, of course, disable wallpaper tinting when the sliver is present in the widget tree. However, since the `SliverToolbarPage` is hidden inside an `IndexedStack`, it is not always visible, so I don’t need to always disable wallpaper tinting — only when the toolbar actually visible on-screen.

For this reason, I needed a way to determine if the `SliverToolbarPage` was currently visible, which I did by replacing `pages` with a list of builders that can be informed if the widget they build is currently visible.

Had I not done this, wallpaper tinting would always be disabled since the `SliverToolbarPage` is always present in the widget tree.

The solution is supposed to be temporary and be reverted once custom backdrop filters are supported in Flutter, but until then, the only other solution I can think of is to use [flutter_bloc](https://pub.dev/packages/flutter_bloc) to manage the index’ state.

Do you think that would be a cleaner solution?

Reply:
Fair enough.

No, I don't think we need to use bloc for this. We can leave it.